packages/flame_devtools/lib/widgets/component_tree.dart

```diff
@@ -1,148 +1,115 @@
-                  ),
-                  if (value.hasData)
-                    Expanded(
-                      child: SingleChildScrollView(
-                        child: TreeView.simple(
-                          showRootNode: false,
-                          shrinkWrap: true,
-                          indentation: const Indentation(
-                            color: Colors.blue,
-                            style: IndentStyle.roundJoint,
-                          ),
-                          onTreeReady: (controller) =>
-                              controller.expandAllChildren(controller.tree),
-                          padding: const EdgeInsets.only(left: 20),
-                          expansionIndicatorBuilder: (context, node) =>
-                              node.isLeaf
-                                  ? NoExpansionIndicator(tree: node)
-                                  : ChevronIndicator.rightDown(
-                                      tree: node,
-                                      alignment: Alignment.centerLeft,
-                                    ),
-                          builder: (context, node) {
-                            return Padding(
-                              padding: node.isLeaf
-                                  ? EdgeInsets.zero
-                                  : const EdgeInsets.only(left: 20),
-                              child: ListTile(
-                                key: Key(
-                                  node.data?.id.toString() ?? node.key,
-                                ),
-                                selected: node == _selectedTreeNode,
-                                selectedColor: theme.colorScheme.primary,
-                                title: Text(node.data!.name),
-                                subtitle: Text(node.data!.id.toString()),
-                                onTap: () {
-                                  return setState(
-                                    () => _selectedTreeNode = node,
-                                  );
-                                },
-                              ),
-                            );
-                          },
-                          tree: _tree,
-                        ),
+    return RoundedOutlinedBorder(
+      child: Column(
+        children: [
+          AreaPaneHeader(
+            title: Row(
+              children: [
+                Text(
+                  'Component Tree ($componentCount components)',
+                  style: theme.textTheme.titleSmall,
+                ),
+                IconButton(
+                  icon: const Icon(Icons.refresh),
+                  iconSize: 18,
+                  alignment: Alignment.center,
+                  onPressed: loader.isLoading
+                      ? null
+                      : () => ref.refresh(componentTreeLoaderProvider),
+                ),
+              ],
+            ),
+          ),
+          Expanded(
```

Comment:
no value.hasData anymore?

Reply:
Nah, now it is showing the tree all the time (even an invisible root node), which makes it more efficient.
The loader doesn't really give anything since the local network requests are so fast that it just blinks.